src/io/file_descriptor.cr

```diff
@@ -38,10 +38,10 @@ class IO::FileDescriptor < IO
     # Figure out the terminal TTY name. If ttyname fails we have a non-tty, or something strange.
     path = uninitialized UInt8[256]
     ret = LibC.ttyname_r(fd, path, 256)
-    return new(fd) unless ret == 0
+    return new(fd).tap(&.flush_on_newline=(true)) unless ret == 0
 
     clone_fd = LibC.open(path, LibC::O_RDWR)
-    return new(fd) if clone_fd == -1
+    return new(fd).tap(&.flush_on_newline=(true)) if clone_fd == -1
```

Comment:
Do we want a comment explaining why the flush is happening? It was obviously not obvious enough the first time :)

Reply:
Good idea! I added a couple more commits explaining this. Thank you!